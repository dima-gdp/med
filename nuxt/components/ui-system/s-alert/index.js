import SAlert from './s-alert'

export default SAlert
