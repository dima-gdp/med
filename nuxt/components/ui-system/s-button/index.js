import SButton from './s-button'

export default SButton
