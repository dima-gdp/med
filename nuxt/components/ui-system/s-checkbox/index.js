import SCheckbox from './s-checkbox'

export default SCheckbox
