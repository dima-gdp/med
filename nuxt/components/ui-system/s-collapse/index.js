import SCollapse from './s-collapse'
import SCollapseItem from './s-collapse-item'

export { SCollapse, SCollapseItem }
