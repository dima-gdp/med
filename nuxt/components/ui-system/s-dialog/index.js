import SDialog from './s-dialog'

export default SDialog
