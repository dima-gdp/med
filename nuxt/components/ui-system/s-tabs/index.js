import STabs from './s-tabs'
import STabItem from './s-tab-item'

export { STabs, STabItem }
