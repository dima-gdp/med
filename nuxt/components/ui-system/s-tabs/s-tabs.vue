<template>
  <div :class="classTabs">
    <div :class="`${classComponent}__nav`">
      <ul :class="`${classComponent}__nav-list`">
        <li
          v-for="(tab, index) in tabs"
          :key="index"
          class="sp-tabs"
          :class="classTabItem(index)"
          @click="selectTab(index)"
        >
          {{ tab.componentInstance.title }}
        </li>
      </ul>
    </div>
    <div :class="`${classComponent}__content`">
      <slot></slot>
    </div>
  </div>
</template>
<script>
import { reactive, provide, toRefs, onMounted, computed, watch } from '@nuxtjs/composition-api'
export default {
  name: 'STabs',

  props: {
    value: { type: Number, default: 0 },
  },

  setup(props, ctx) {
    const classComponent = 's-tabs'

    const state = reactive({
      selected: 0,
      tabs: [],
      count: 0,
    })

    provide('TabsProvider', state)

    watch(() => props.value, (val) => {
      state.selected = val
    })

    const classTabs = computed(() => [
      { [`${classComponent}`]: true },
      { [`${classComponent}_checked`]: props.value },
      { [`${classComponent}--disabled`]: props.disabled },
    ])

    const classTabItem = (index) => {
      return [
        { [`${classComponent}__nav-item`]: true },
        { [`${classComponent}__nav-item--active`]: index === state.selected },
      ]
    }

    const selectTab = (i) => {
      ctx.emit('input', i)
    }

    onMounted(() => {
      if (ctx.slots.default) {
        state.tabs = ctx.slots
          .default()
          .filter((child) => child.componentOptions?.tag === 'STabItem')
      }
    })

    return {
      ...toRefs(state),
      classComponent,
      classTabs,
      classTabItem,
      selectTab,
    }
  },
}
</script>
<style lang="scss">
@import 's-tabs';
</style>
