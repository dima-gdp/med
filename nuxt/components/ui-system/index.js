import SButton from './s-button'
import { SMenu, SMenuItem } from './s-menu'
import SAlert from './s-alert'
import SInput from './s-input'
import SInputPin from './s-input-pin'
import { SCollapse, SCollapseItem } from './s-collapse'
import SDialog from './s-dialog'
import SCheckbox from './s-checkbox'
import SSelect from './s-select'
import SSpin from './s-spin'
import { SRadio, SRadioGroup, SRadioButton } from './s-radio'
import { STabs, STabItem } from './s-tabs'
import SErrorMessage from './s-error-message'

export {
  SButton,
  SMenu,
  SMenuItem,
  SAlert,
  SCollapse,
  SCollapseItem,
  SDialog,
  SInput,
  SInputPin,
  SCheckbox,
  STabs,
  STabItem,
  SSelect,
  SSpin,
  SErrorMessage,
  SRadio,
  SRadioGroup,
  SRadioButton,
}
