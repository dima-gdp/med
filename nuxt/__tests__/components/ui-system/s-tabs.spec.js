import { mount } from '@vue/test-utils'
import { STabs } from '~/components/ui-system/s-tabs'

describe('ui-system/s-tabs', () => {
  it('s-tabs отображается', () => {
    const wrapper = mount(STabs)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
