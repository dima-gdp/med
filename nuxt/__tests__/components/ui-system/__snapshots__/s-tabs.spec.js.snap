// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-system/s-tabs s-tabs отображается 1`] = `
<div class="s-tabs">
  <div class="s-tabs__nav">
    <ul class="s-tabs__nav-list"></ul>
  </div>
  <div class="s-tabs__content"></div>
</div>
`;
